### PR TITLE
NMS-15431: Mina SSH Monitor

### PR DESCRIPTION
**Pull request by RangerRick**
This PR extends the work @roskens did in [his pr](https://github.com/OpenNMS/opennms/pull/5793) to include changes related to my recent work upgrading and normalizing our Mina SSH dependencies to address some security issues.

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15431

### CODE REVIEW

File: docs/modules/reference/pages/service-assurance/monitors/MinaSshMonitor.adoc [hunk 1]
```diff
@@ -0,0 +1,80 @@
+
+= MinaSshMonitor
+
+The MinaSshMonitor tests the availability of an SSH service.
+During the poll, it attempts to connect on the specified port and then authenticate in.
+If the authentication is successful, then the service is considered up.
+Optionally, a remote command to execute can be provided and if that is successful, the service is considered up.
+
+== MinaSshMonitor vs SshMonitor
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/MinaSshMonitor.adoc:9]:
```suggestion
== MinaSshMonitor vs. SshMonitor
```

File: docs/modules/reference/pages/service-assurance/monitors/MinaSshMonitor.adoc [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+The MinaSshMonitor tests connecting and authenticating to a node via SSH.
+The SshMonitor only tests that the SSH port is accepting connections.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/MinaSshMonitor.adoc:12]:
```suggestion
The xref:service-assurance/monitors/SshMonitor.adoc[SshMonitor] only tests that the SSH port is accepting connections.
```

File: docs/modules/reference/pages/service-assurance/monitors/MinaSshMonitor.adoc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+| password
+| A password for username to authenticate with on the node.
+| n/a
+
+| identity-file
+| An identity-file to use for passwordless authenticatition.
+| n/a
+
+| remote-command
+| A command to execute after authentication is successful on the node. The output is ignored, and any execution errors will mark the service as down.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/MinaSshMonitor.adoc:45]:
```suggestion
| A command to execute after authentication is successful on the node.
The output is ignored, and any execution errors will mark the service as down.
```

**Comment by RangerRick** [comment]
> minor updates in the docs. Also, is there a corresponding detector?

The "regular" (port-based) `SshDetector` should be fine for detecting in this case, although the default configs aren't modified to use it.
Outside of the scope for this change, but we may want to deprecate the `SshMonitor` and change the default configs to use this (with an `SshDetector` configured to match). I'll talk to @jeffgdotorg about it.

**Comment by mmahacek** [comment]
FWIW, the SsshDetector does not appear to be documented.

**Comment by Bonrob2** [comment]
> FWIW, the SsshDetector does not appear to be documented.

It's listed in NMS-9014, but we haven't identified it as one to be documented.

**Review by Bonrob2 — APPROVED** [review]
docs LGTM. Thanks!

**Pull request merged**